### PR TITLE
osdc/Filer: truncate large file party by party

### PR DESCRIPTION
**Pull request by ukernel**
Fixes: http://tracker.ceph.com/issues/19755
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>

### CODE REVIEW

**Comment by batrick** [comment]
cephfs integration test run: http://pulpito.ceph.com/pdonnell-2017-05-09_17:36:48-fs-wip-pdonnell-integration-distro-basic-smithi/

File: src/osdc/Filer.cc [hunk 1]
```diff
@@ -377,3 +377,108 @@ void Filer::_do_purge_range(PurgeRange *pr, int fin)
...
+
+    uint64_t offset = tr->offset + tr->length - len;
+    Striper::file_to_extents(cct, tr->ino, &tr->layout, offset, len, 0, extents);
+    tr->uncommitted += extents.size();
+    tr->length -= len;
+  }
+
+  trl.unlock();
+
+  // Issue objecter ops outside tr->lock to avoid lock dependency loop
```
Comment (jcsp) [inline src/osdc/Filer.cc]: Since we have dropped the lock here, why is it still necessary to use a C_OnFinisher below?

**Comment by jcsp** [comment]
This passes tests, just that one comment about whether the use of finisher is really needed.

File: src/osdc/Filer.cc (hunk 1 above)
Comment (ukernel) [inline src/osdc/Filer.cc]: see http://tracker.ceph.com/issues/10229

**Pull request merged**